modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshotsManager.java

```diff
@@ -65,41 +91,69 @@ public MessagingService messagingService() {
 
     @Override
     public void start() {
-        messagingService.addMessageHandler(TableMessageGroup.class, this::messageHandler);
+        executor = Executors.newFixedThreadPool(4, new NamedThreadFactory("outgoing-snapshots", LOG));
```

Comment:
I just did not want to start the pool before the component is started. The field is `volatile` because `start()` might be called from a thread different from the thread where the object is instantiated.

`4` is a first guess. This should probably be configured. The PR is already pretty bloated. Let's leave it as is and continue in IGNITE-17935.

Reply:
Ok